resolver-dns/src/main/java/io/netty/resolver/dns/DnsServerAddressStream.java

```diff
@@ -29,4 +29,16 @@
      * Retrieves the next DNS server address from the stream.
      */
     InetSocketAddress next();
+
+    /**
+     * Get the number of times {@link #next()} will return a distinct element before repeating or terminating.
+     * @return the number of times {@link #next()} will return a distinct element before repeating or terminating.
+     */
+    int size();
+
+    /**
+     * Clone this object. The result of this should be able to be independently iterated over via {@link #next()}.
+     * @return A clone of this object
+     */
+    DnsServerAddressStream clone();
```

Comment:
@Scottmitch maybe we should use another method name to not "conflict" with `Clonable` ?